app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -185,7 +185,8 @@ def regenerate_identity_certificates
     end
     param :id, String, :desc => N_("UUID of the system"), :required => true
     def enabled_repos
-      repos_params = params['enabled_repos'] rescue raise(HttpErrors::BadRequest, _("Expected attribute is missing:") + " enabled_repos")
+      fail(HttpErrors::BadRequest, _("Expected attribute is missing:") + " enabled_repos") if params['enabled_repos'].blank?
```

Comment:
The way this is written will prevent enabled repos from being updated even if the 'enabled_repos' param is empty. I think we might want it to update so long as there is any value for the param - even an empty array.

I recommend writing this code as follows:

```
fail(HttpErrors::BadRequest, _("Expected attribute is missing:") + " enabled_repos") unless params['enabled_repos']

repos_params = params['enabled_repos'].try(:[], 'repos') || []
```

Reply:
This might be better, and definitely more clean

```
repos = params.dig(:enabled_repos, :repos)
fail() if repos.nil?
```